packages/cadl-ranch-specs/http/azure/core/traits/mockapi.ts

```diff
@@ -17,8 +18,15 @@ Scenarios.Azure_Core_Traits_smokeTest = passOnSuccess(
       throw new ValidationError("Expected path param id=1", "1", req.params.id);
     }
     req.expect.containsHeader("foo", "123");
-    req.expect.containsHeader("if-match", '"valid"');
-    req.expect.containsHeader("if-none-match", '"invalid"');
+    const if_none_match = req.headers["if-none-match"];
+    const if_match = req.headers["if-match"];
+    if (if_none_match !== '"invalid"' && if_match !== '"valid"') {
+      throw new ValidationError(
+        `Expected header "if-none-match" equals "invalid" but got ${if_none_match} or "if-match" equals "valid" but got ${if_match}`,
```

Comment:
It is unreasonable to set if-match and if-none-match at the same time. And the change doesn't break existing test.
Add @weidongxu-microsoft @tadelesh @chunyu3 @qiaozha to review